autogpt/llm/base.py

```diff
@@ -95,19 +116,18 @@ def insert(self, index: int, *messages: Message):
             self.messages.insert(index, message)
 
     @classmethod
-    def for_model(cls, model_name: str, messages: list[Message] | ChatSequence = []):
+    def for_model(
```

Comment:
No suggestion here, just some questions: 

- Why does the ChatSequence have any knowledge of a model?  
- Don't we want to be able to potentially use the same chat sequence with multiple models?

Reply:
> Why does the ChatSequence have any knowledge of a model?

I designed the `ChatSequence` with usage as a prompt object in mind.
With how this has evolved now, maybe it would be better to make a subclass `ChatPrompt`, with additional `model` and `functions` attributes over the base class.

Reply:
> Don't we want to be able to potentially use the same chat sequence with multiple models?

Maybe in the future, but I currently see no use case for that and accommodating for that now feels like premature optimization.

Reply:
> > Why does the ChatSequence have any knowledge of a model?
> 
> I designed the `ChatSequence` with usage as a prompt object in mind. As you can see, it's used in the `token_length` computed property. With how this has evolved now, we could make a subclass `ChatPrompt` or `ChatModelSequence`, with additional `model` and `functions` attributes over the base class.

I think this is worth talking through more, but doesn't need resolution for this PR.